templates/e2e-llm-workflows/README.ipynb

```diff
@@ -512,12 +491,52 @@
     }
    ],
    "source": [
+    "import anyscale\n",
+    "from ray.data.datasource import FilenameProvider\n",
+    "from rich import print as rprint\n",
+    "from typing import Dict\n",
+    "\n",
+    "import os\n",
+    "\n",
+    "class CustomFilenameProvider(FilenameProvider):\n",
+    "    def __init__(self, custom_filename: str):\n",
+    "        self.custom_filename = custom_filename\n",
+    "    \n",
+    "    def get_filename_for_block(self, *args, **kwargs) -> str:\n",
+    "        return self.custom_filename\n",
+    "\n",
     "# Write to cloud storage\n",
-    "fs = pyarrow.fs.PyFileSystem(pyarrow.fs.FSSpecHandler(S3FileSystem()))\n",
-    "storage_path = os.environ['ANYSCALE_ARTIFACT_STORAGE'][len('s3://'):]\n",
-    "ft_train_ds.write_json(f'{storage_path}/viggo/train.jsonl', filesystem=fs)\n",
-    "ft_val_ds.write_json(f'{storage_path}/viggo/val.jsonl', filesystem=fs)\n",
-    "ft_test_ds.write_json(f'{storage_path}/viggo/test.jsonl', filesystem=fs)"
+    "temp_path = '/tmp/e2e_llm/viggo'\n",
+    "os.makedirs(temp_path, exist_ok=True)\n",
+    "config_fields: Dict[str, str] = {}\n",
+    "for dataset, filename, field_name in [\n",
+    "    (ft_train_ds, 'train.jsonl', \"train_path\"),\n",
+    "    (ft_val_ds, 'val.jsonl', \"valid_path\"),\n",
+    "    (ft_test_ds, 'test.jsonl', \"test_path\")\n",
+    "]:\n",
+    "    dataset.write_json(temp_path, filename_provider=CustomFilenameProvider(filename))\n",
+    "    def get_full_name():\n",
+    "        full_name = os.environ.get(\"ANYSCALE_USERNAME\", \"default\")\n",
+    "        full_name = \"_\".join(full_name.split())  # Join the full name with underscores\n",
+    "        full_name = full_name.lower()  # Lowercase the full name\n",
+    "        return full_name\n",
+    "    \n",
+    "    # full_name/e2e_llm/viggo/train.jsonl\n",
+    "    dataset_name = f\"{get_full_name()}/e2e_llm/viggo/{filename}\"\n",
```

Comment:
I'd set the name to something more user-friendly like viggo-training instead of a file path

Reply:
this would be fine for ray summit training day

but for regular use case, it would use the default project of the cloud they're in, so users in the same cloud could be overriding (uploading new versions) of this `viggo-training` if they're working with the template at the same time

Reply:
maybe we can call it viggo-training-{user}

Reply:
sure, that seems reasonable
```py
    dataset_name = f"viggo-training-{get_full_name()}",
```

keep in mind - they can also modify the dataset name to something of their own choice

Reply:
i see, i'm probably overthinking it then, i'll do `viggo/{filename}`

i think a path-like structure is fine b/c our models are named `meta.../llama...`, so they're used to this structure already

Reply:
fixed in https://github.com/anyscale/templates/pull/313/commits/d87083f703e3316dd69bba3493f9d966b5dd4dd2